packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/AppConfigurationTests.kt

```diff
@@ -394,6 +397,37 @@ class AppConfigurationTests {
 //        looperThread.testComplete()
 //    }
 
+    @Test
+    fun encryptionKey() {
+        val key = TestHelper.getRandomKey()
+        val config = AppConfiguration.Builder(APP_ID)
+            .encryptionKey(key)
+            .build()
+
+        assertContentEquals(key, config.encryptionKey)
+    }
+
+    @Test
+    fun encryptionKey_isCopy() {
+        val key = TestHelper.getRandomKey()
+        val config = AppConfiguration.Builder(APP_ID)
+            .encryptionKey(key)
+            .build()
+
+        assertNotSame(key, config.encryptionKey)
+    }
+
+    @Test
+    fun encryptionKey_wrongLength() {
```

Comment:
We are missing tests that actually verify that the user data Realm has been encrypted. That can be achieved by creating a local `RealmConfiguration` that points to it and then verify that you need the encryption key to open it.

See e.g. the similar tests in Realm Java here https://github.com/realm/realm-java/blob/master/realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/AppTests.kt#L273

That test uses a DynamicRealm, in Kotlin I think you can get away with just using an empty schema..

Reply:
The metadata realm should be located in `<syncRoot>/<appId>/metadata/sync_metadata.realm`

Reply:
Thanks Christian! 👍